app/Support/Cache/CacheKey.php

```diff
@@ -0,0 +1,82 @@
+    public static function buildUserCanTicketCacheKey(string $userName): string
+    {
+        return self::buildNormalizedUserCacheKey($userName, "canTicket");
+    }
+
+    public static function buildUserCardDataCacheKey(string $userName): string
+    {
+        return self::buildNormalizedUserCacheKey($userName, "card-data");
+    }
+
+    /**
+     * @param string $userName the name of the user for which the cache key is being constructed
+     * @param int $gameID the target game ID for the unlocks data
+     * @param int $flags The type of achievement flag which should correspond to values in the `AchievementType` enum.
+     *                   3 for 'OfficialCore' (default), 5 for 'Unofficial'.
+     */
+    public static function buildUserGameUnlocksCacheKey(string $userName, int $gameID, int $flags = 3): string
+    {
+        return self::buildNormalizedUserCacheKey($userName, "gameUnlocks", [$gameID, $flags]);
+    }
+
+    /**
+     * @param string $userName the name of the user for which the cache key is being constructed
+     * @param int $rankType The type of the rank which should correspond to values in the `RankType` enum.
+     *                      1 for 'Hardcore' (default), 2 for 'Softcore', 3 for 'TruePoints'.
+     */
+    public static function buildUserRankCacheKey(string $userName, int $rankType = 1): string
+    {
+        $rankTypeParam = match ($rankType) {
+            default => 'hardcore',
+            2 => 'softcore',
+            3 => 'truepoints',
+        };
+
+        return self::buildNormalizedUserCacheKey($userName, "rank", [$rankTypeParam]);
+    }
+
+    public static function buildUserRecentGamesCacheKey(string $userName): string
+    {
+        return self::buildNormalizedUserCacheKey($userName, "recentGames");
+    }
+
+    /**
+     * Constructs a normalized user cache key.
+     *
+     * This function creates a cache key for a specific user with a specified kind and optional parameters.
+     * The username is case-corrected to ensure superfluous cache entries for the same user are not created.
+     * Cache keys follow the format: "user:{lowercase username}:{key kind}:{optional colon-separated parameters}".
+     *
+     * Use the function like this:
+     * ```php
+     * $cacheKey = CacheKey::constructNormalizedUserCacheKey("UserName", "gameUnlocks", [$gameID, $flags]);
+     * ```
+     * This generates a cache key like: "user:username:gameUnlocks:{gameID}:{flags}".
+     *
+     * @param string $userName the name of the user for which the cache key is constructed
+     * @param string $keyKind the kind of the cache key
+     * @param array  $params optional parameters for the cache key
+     *
+     * @return string the constructed cache key
+     */
+    private static function buildNormalizedUserCacheKey(string $userName, string $keyKind, array $params = []): string
+    {
+        $normalizedUserName = strtolower($userName);
+
+        $cacheKey = "user:$normalizedUserName:$keyKind";
```

Comment:
It seems we have quite the mix of cases in our code already - initially we went for kebap-case for cache keys (`card-data`), now we have some camelCase in there, too.
I think we should at least normalize cache keys by lowercasing it fully. Not perfect but one option less to mistype it. What do you think?

Reply:
I agree it makes sense to keep one form of casing. At the same time, I didn't want to break any existing caching that may already be set up. What do you think would be best?

Reply:
As long as the changes are consistent between reads/writes/deletes, it doesn't matter what the old keys were - they'll be purged when they expire.

Reply:
Cases have been corrected in https://github.com/RetroAchievements/RAWeb/pull/1659/commits/01bbb1c2d5a090e2536286aaa68e04990aee8883.